### PR TITLE
ModuleInfo for Entrypoint in LoadStrategy

### PR DESCRIPTION
**Pull request by grouma**
We would like to customize the URIs contained in the dependency map. Internally we would like to use absolute file paths while externally with `package:build_runner` we would like to use `HTTP` URIs. Add `moduleInfoForEntrypoint` as a part of the `LoadStrategy` to support his customization.

### CODE REVIEW

File: dwds/lib/src/loaders/build_runner_require.dart [hunk 1]
```diff
@@ -31,6 +32,7 @@ class BuildRunnerRequireStrategyProvider {
         _serverPathForModule,
         _sourceMapPathForModule,
         _serverPathForAppUri,
+        _moduleInfoForProvider,
```
Comment (annagrin) [inline dwds/lib/src/loaders/build_runner_require.dart:35]: Should this be _moduleInfoForEntryPoint to match other names?

File: dwds/lib/src/loaders/build_runner_require.dart [hunk 2]
```diff
@@ -82,4 +84,21 @@ class BuildRunnerRequireStrategyProvider {
     }
     return null;
   }
+
+  Future<Map<String, ModuleInfo>> _moduleInfoForProvider(
```
Comment (annagrin) [inline dwds/lib/src/loaders/build_runner_require.dart:88]: Should it be just `_moduleInfo` to match the naming pattern?

File: dwds/lib/src/loaders/frontend_server_require.dart [hunk 3]
```diff
@@ -53,4 +56,19 @@ class FrontendServerRequireStrategyProvider {
     }
     return null;
   }
+
+  Future<Map<String, ModuleInfo>> _moduleInfoForProvider(
```
Comment (annagrin) [inline dwds/lib/src/loaders/frontend_server_require.dart:60]: _moduleInfo?

File: dwds/lib/src/loaders/legacy.dart [hunk 4]
```diff
@@ -22,6 +22,15 @@ class LegacyStrategy extends LoadStrategy {
...
       _moduleForServerPath;
 
+  /// Returns a map from module id to module info.
+  ///
+  /// For example:
+  ///
+  ///   web/main -> {main.ddc.full.dill, main.ddc.dill}
+  ///
+  final Future<Map<String, ModuleInfo>> Function(
+      MetadataProvider metadataProvider) _moduleInfoForProvider;
```
Comment (annagrin) [inline dwds/lib/src/loaders/legacy.dart:32]: same here

File: dwds/lib/src/loaders/require.dart [hunk 5]
```diff
@@ -114,6 +114,15 @@ class RequireStrategy extends LoadStrategy {
...
 
+  /// Returns a map from module id to module info.
+  ///
+  /// For example:
+  ///
+  ///   web/main -> {main.ddc.full.dill, main.ddc.dill}
+  ///
+  final Future<Map<String, ModuleInfo>> Function(
+      MetadataProvider metadataProvider) _moduleInfoForProvider;
+
```
Comment (annagrin) [inline dwds/lib/src/loaders/require.dart:125]: same here

**Review by annagrin — APPROVED** [review]
LGTM with some nits

File: dwds/lib/src/loaders/build_runner_require.dart (hunk 1 above)
Comment (grouma) [reply dwds/lib/src/loaders/build_runner_require.dart:35]: No. This method uses a provider and not the entrypoint.

File: dwds/lib/src/loaders/build_runner_require.dart (hunk 2 above)
Comment (grouma) [reply dwds/lib/src/loaders/build_runner_require.dart:88]: It's already consistent with the expected input.

**Pull request merged**